### PR TITLE
babl 0.1.72

### PR DESCRIPTION
**Pull request by zachauten**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by dawidd6** [comment]
They switched to `meson`.

**Comment by jonchang** [comment]
```
[79/181] Linking target babl/libbabl-0.1.0.dylib.
FAILED: babl/libbabl-0.1.0.dylib 
clang  -o babl/libbabl-0.1.0.dylib 'babl/4170c83@@babl-0.1@sha/babl-cache.c.o' 'babl/4170c83@@babl-0.1@sha/babl-component.c.o' 'babl/4170c83@@babl-0.1@sha/babl-conversion.c.o' 'babl/4170c83@@babl-0.1@sha/babl-core.c.o' 'babl/4170c83@@babl-0.1@sha/babl-cpuaccel.c.o' 'babl/4170c83@@babl-0.1@sha/babl-db.c.o' 'babl/4170c83@@babl-0.1@sha/babl-extension.c.o' 'babl/4170c83@@babl-0.1@sha/babl-fish-path.c.o' 'babl/4170c83@@babl-0.1@sha/babl-fish-reference.c.o' 'babl/4170c83@@babl-0.1@sha/babl-fish-simple.c.o' 'babl/4170c83@@babl-0.1@sha/babl-fish.c.o' 'babl/4170c83@@babl-0.1@sha/babl-format.c.o' 'babl/4170c83@@babl-0.1@sha/babl-hash-table.c.o' 'babl/4170c83@@babl-0.1@sha/babl-icc.c.o' 'babl/4170c83@@babl-0.1@sha/babl-image.c.o' 'babl/4170c83@@babl-0.1@sha/babl-internal.c.o' 'babl/4170c83@@babl-0.1@sha/babl-introspect.c.o' 'babl/4170c83@@babl-0.1@sha/babl-list.c.o' 'babl/4170c83@@babl-0.1@sha/babl-memory.c.o' 'babl/4170c83@@babl-0.1@sha/babl-model.c.o' 'babl/4170c83@@babl-0.1@sha/babl-mutex.c.o' 'babl/4170c83@@babl-0.1@sha/babl-palette.c.o' 'babl/4170c83@@babl-0.1@sha/babl-polynomial.c.o' 'babl/4170c83@@babl-0.1@sha/babl-ref-pixels.c.o' 'babl/4170c83@@babl-0.1@sha/babl-sampling.c.o' 'babl/4170c83@@babl-0.1@sha/babl-sanity.c.o' 'babl/4170c83@@babl-0.1@sha/babl-space.c.o' 'babl/4170c83@@babl-0.1@sha/babl-trc.c.o' 'babl/4170c83@@babl-0.1@sha/babl-type.c.o' 'babl/4170c83@@babl-0.1@sha/babl-util.c.o' 'babl/4170c83@@babl-0.1@sha/babl-version.c.o' 'babl/4170c83@@babl-0.1@sha/babl.c.o' -Wl,-dead_strip_dylibs -Wl,-undefined,error -Wl,-headerpad_max_install_names -shared -install_name @rpath/libbabl-0.1.0.dylib -compatibility_version 0 -current_version 0 -Wl,-force_load babl/base/libbabl_base.a -Wl,--version-script,babl.map -lm -ldl /usr/local/Cellar/little-cms2/2.9/lib/liblcms2.dylib -Wl,-headerpad_max_install_names -Wl,-rpath,/usr/local/Cellar/little-cms2/2.9/lib
ld: unknown option: --version-script
```

Only GNU `ld` supports `--version-script`: https://www.gnu.org/software/gnulib/manual/html_node/LD-Version-Scripts.html

Probably have to remove this from the meson build file.

**Comment by jonchang** [comment]
This patch might be useful:

https://gitlab.gnome.org/GNOME/babl/commit/a04535b0964d2bd40d08e4b3d16ff27b6ee7262c

**Comment by zachauten** [comment]
@BrewTestBot test this please

**Comment by zachauten** [comment]
Still having some trouble figuring this one out. Current error message:

```
[178/181] Generating Babl-0.1.gir with a custom command.
g-ir-scanner: link: clang -o /private/tmp/babl-20191126-81382-izs21u/babl-0.1.72/build/tmp-introspectf6cu306u/Babl-0.1 /private/tmp/babl-20191126-81382-izs21u/babl-0.1.72/build/tmp-introspectf6cu306u/Babl-0.1.o -L. -Wl,-rpath,. -L/private/tmp/babl-20191126-81382-izs21u/babl-0.1.72/build/babl -Wl,-rpath,/private/tmp/babl-20191126-81382-izs21u/babl-0.1.72/build/babl -L/usr/local/Cellar/little-cms2/2.9/lib -Wl,-rpath,/usr/local/Cellar/little-cms2/2.9/lib -L/usr/local/Cellar/little-cms2/2.9/lib -Wl,-rpath,/usr/local/Cellar/little-cms2/2.9/lib -lbabl-0.1 -lm -ldl -llcms2 -L/usr/local/Cellar/glib/2.62.2/lib -L/usr/local/opt/gettext/lib -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lglib-2.0 -lintl
[179/181] Generating Babl-0.1.typelib with a custom command.
[180/181] Generating index.html.tmp with a meson_exe.py custom command.
../babl/babl-internal.h:214 void babl_log(const char *, ...)()
	WARNING: the babl installation seems broken, no extensions found in queried
BABL_PATH (/private/tmp/babl-20191126-81382-izs21u/babl-0.1.72/build/extensions) this means no SIMD/instructions/special case fast paths and
only slow reference conversions are available, applications might still
run but software relying on babl for conversions will be slow

[181/181] Generating Reference.html with a custom command.
==> ninja install
[0/1] Installing files.
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 129, in run
    return options.run_func(options)
  File "/usr/local/lib/python3.7/site-packages/mesonbuild/minstall.py", line 521, in run
    installer.do_install(datafilename)
  File "/usr/local/lib/python3.7/site-packages/mesonbuild/minstall.py", line 343, in do_install
    self.install_targets(d)
  File "/usr/local/lib/python3.7/site-packages/mesonbuild/minstall.py", line 450, in install_targets
    self.do_copyfile(fname, outname)
  File "/usr/local/lib/python3.7/site-packages/mesonbuild/minstall.py", line 257, in do_copyfile
    shutil.copyfile(from_file, to_file)
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
PermissionError: [Errno 1] Operation not permitted: '/usr/local/lib/libbabl-0.1.0.dylib'
Installing babl/libbabl-0.1.0.dylib to /usr/local/lib
FAILED: meson-install 
/usr/local/Cellar/meson/0.52.0/libexec/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.

```

**Comment by Bo98** [comment]
> ```
> Installing babl/libbabl-0.1.0.dylib to /usr/local/lib
> ```

You'll have to make it not do that and install to the Cellar instead (given by `prefix`). If meson supports `--prefix=#{prefix}` then do that.

**Comment by zachauten** [comment]
@BrewTestBot test this please

**Comment by tschoonj** [comment]
I think you can drop `libtool` from the dependencies, and perhaps `intltool` as well.

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

File: Formula/babl.rb [hunk 1]
```diff
@@ -13,16 +13,24 @@ class Babl < Formula
     sha256 "3cd4d6d3cc86dc1bc2b8411bf4ffb911df58458d54fc938730fdede11587c624" => :sierra
   end
 
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
   depends_on "libtool" => :build
+  depends_on "little-cms2" => :build
```
Comment (fxcoudert) [inline Formula/babl.rb]:
```suggestion
  depends_on "little-cms2"
```

`little-cms2` is a runtime dependency. Place it after `gobject-introspection`

File: Formula/gegl.rb [hunk 2]
```diff
@@ -16,12 +15,14 @@ class Gegl < Formula
...
-    depends_on "autoconf" => :build
-    depends_on "automake" => :build
     depends_on "libtool" => :build
   end
 
+  depends_on "cmake" => :build
   depends_on "intltool" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+
```
Comment (fxcoudert) [inline Formula/gegl.rb]: Remove empty line.

File: Formula/gegl.rb [hunk 3]
```diff
@@ -33,16 +34,12 @@ class Gegl < Formula
...
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
-                          "--prefix=#{prefix}",
-                          "--disable-docs",
-                          "--without-cairo",
-                          "--without-jasper",
-                          "--without-umfpack",
-                          "--without-libspiro"
-    system "make", "install"
+    ENV.prepend_path "PKG_CONFIG_PATH", "/usr/local/opt/lcms2/lib/pkgconfig"
```
Comment (fxcoudert) [inline Formula/gegl.rb]: Add it as a dependency if you want that.

File: Formula/gegl.rb [hunk 4]
```diff
@@ -33,16 +34,12 @@ class Gegl < Formula
...
+    system "meson", "build", "--prefix=#{prefix}"
```
Comment (fxcoudert) [inline Formula/gegl.rb]:
```
19:25:26 meson.build:150:0: ERROR: Could not generate cargs for babl:
19:25:26 Package lcms2 was not found in the pkg-config search path.
19:25:26 Perhaps you should add the directory containing `lcms2.pc'
19:25:26 to the PKG_CONFIG_PATH environment variable
19:25:26 Package 'lcms2', required by 'babl', not found
```

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by ryan-robeson** [comment]
Possibly relevant:
https://gitlab.gnome.org/GNOME/gegl/issues/214

**Comment by zachauten** [comment]
@ryan-robeson Thanks! That's the same error i've been getting, I'll try adding p2c.

**Comment by zachauten** [comment]
@Homebrew/core The thread @ryan-robeson linked to suggests this is failing due to a bug in meson on macos, and the way gegl installs the dependency poly2tri-c. What's the best course of action here?

**Comment by fxcoudert** [comment]
Either the meson bug gets fixed, or gegl introduces a work-around (which they seem to be doing).

**Pull request closed without merging**